### PR TITLE
update dvc pipelines by hand

### LINKED ISSUE #230: how to: "update your pipeline by editing DVC-files"
See https://github.com/iterative/dvc/issues/1800#issuecomment-478994887 .

Might be worth publishing under "Best Practices" https://github.com/iterative/dvc.org/issues/72 .

### PR DESCRIPTION
**Pull request by Naba7**
Fixes #230

### CODE REVIEW

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# How to update your pipeline by editing dvc files
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: - DVC-files
- simpler name - `Updating pipelines`. Especially considering that you also mention `dvc move`.
- update sidebar.json to connect this new section
- remove .share-data.swp from the PR

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+> 1)using `dvc move` to edit file paths of outputs or dependencies.
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: some introduction is required above. I don't quite understand what's happening here. Why it's `>`-ed. What is it about. Etc.

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+> 2)editting by hand the outputs paths.
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: space missing. Use capital letters to start list items if they contain full sentences.

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+> 3)`dvc run` to make edits in the dependency files and `dvc commit` to save
+> changes.
+
+If your source file is a data file, then use dvc move:
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: dvc move -> `dvc move`

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+`dvc move` is used if a file or a directory has been added to DVC with `dvc 
+add`. The source file or directory is moved to its destination path, renamed 
+and corresponding DVC-file is updated.
+
+If your source file isn't data file then use `dvc run`. Then add the new file
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: I'm not sure I understand this ... why `dvc add` after `dvc run`?

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+with `dvc add`. If you check your file using `git mv` then it shows that your
+path isn't updated. So, using `git mv` is not an option.
+
+`dvc run` is used to create a dvc pipeline. It can be shown in a connected
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: dvc -> DVC

Comment (shcheklein) [reply static/docs/user-guide/update-DVC-pipelines.md]: actually it's clear that it's DVC pipeline, just pipeline is enough.

Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: A lot of description of `dvc run` itself, not that much information about updating the pipeline.

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 2 above)
Comment (ryokugyu) [reply static/docs/user-guide/update-DVC-pipelines.md]: hey, @Naba7  i will suggest that in introduction first describe why we need to update DVC-files?

for example: in `dvc run` we forget to mention any output file.

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 3 above)
Comment (ryokugyu) [reply static/docs/user-guide/update-DVC-pipelines.md]: also, editting -> editing

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 2 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: okay. working on it

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 5 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: My bad! I have changed it.

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 7]
```diff
@@ -1,21 +1,46 @@
...
-> 1)using `dvc move` to edit file paths of outputs or dependencies.
-> 2)editting by hand the outputs paths.
-> 3)`dvc run` to make edits in the dependency files and `dvc commit` to save
-> changes.
+If you need to rename or relocate the file paths and/or dependencies of a
+specific DVC-file, then you can 
 
-If your source file is a data file, then use dvc move:
+- use `dvc move` to edit file paths of outputs or dependencies.
+- `dvc run` to make edits in the files and/or dependencies and `dvc commit` to
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: dvc run by default does not require `dvc commit`

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 8]
```diff
@@ -1,21 +1,46 @@
...
+If your source file is a data file, then use `dvc move`:
 
 `dvc move` is used if a file or a directory has been added to DVC with `dvc 
 add`. The source file or directory is moved to its destination path, renamed 
 and corresponding DVC-file is updated.
 
-If your source file isn't data file then use `dvc run`. Then add the new file
-with `dvc add`. If you check your file using `git mv` then it shows that your
-path isn't updated. So, using `git mv` is not an option.
+First add DVC files to get tracked by DVC file-tree. Use `dvc add data.dvc`
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: ?? I'm not sure I understand the first sentence here

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+$ tree .
+
+.
+├── result.out
+├── stages
+│   └── test.dvc
+└── test.txt
+
+
+If you check your file using `git mv` then it shows that your path isn't
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: Again, don't quite understand the logic here

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    path: data.tsv
+
+Then use `dvc move` to rename the files and/or direcories and `dvc move`
+change its location and our initial file is also moved.
+
+If your source file isn't data file then use `dvc run`. Change the location of
+the files and/or directories by hand. It will create a pipeline to show the
+changes made to the connected graph.
+
+$ dvc run -d test.txt -f stages/test.dvc -o result.out \
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: should be a code block?

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
+# Updating pipelines
+
+If you need to rename or relocate the file paths and/or dependencies of a
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: not only dependecies, outputs as well

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 7 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: The changes made by ``dvc run`` are stored in cache and after using ``dvc commit`` the stage is saved. I added it to save the present stage of files and/or dependencies and outputs.

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 8 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: move is not permitted for stages that are not data sources. ``dvc move`` can only relocate data files and their paths and outputs. If the file is a .json or .dvc move can't relocate them.

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 10 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: Yes, My bad

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 12]
```diff
@@ -1,34 +1,50 @@
...
+  `dvc commit` to save changes to cache.
 
-If your source file is a data file, then use `dvc move`:
+If you want to relocate or rename data files and/or directories, dependencies
+and outputs use `dvc move`. `dvc move` is not permitted for stages that are
+not data sources.
+
+```dvc
+$ dvc move output/models/variables/risk_dict.json output/models/risk_dict.json
+?[31mError?[39m: failed to move 'output/models/variables/risk_dict.json' -> 'output/models/risk_dict.json'
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: Is this output from Windows? I looks like a bug to me. @efiop do you know if it's expected?

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 13]
```diff
@@ -1,34 +1,50 @@
 # Updating pipelines
 
-If you need to rename or relocate the file paths and/or dependencies of a
-specific DVC-file, then you can 
+If you need to rename or relocate the file paths and/or dependencies and
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: `relocate the file paths` looks weird. It should be probably just `relocate a DVC-file and/or ... of the DVC-file`

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 14]
```diff
@@ -1,34 +1,50 @@
...
+outputs of a specific DVC-file, then you can
 
 - use `dvc move` to edit file paths of outputs or dependencies.
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: dvc move also moves the file

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 15]
```diff
@@ -1,34 +1,50 @@
...
-- `dvc run` to make edits in the files and/or dependencies and `dvc commit` to
-  save changes.
+- `dvc run` to make edits in the files and/or dependencies and outputs then
+  `dvc commit` to save changes to cache.
 
-If your source file is a data file, then use `dvc move`:
+If you want to relocate or rename data files and/or directories, dependencies
+and outputs use `dvc move`. `dvc move` is not permitted for stages that are
+not data sources.
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: I think some explanation is needed here.

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 16]
```diff
@@ -39,8 +55,10 @@ $ tree .
 ├── stages
 │   └── test.dvc
 └── test.txt
+```
 
-
-If you check your file using `git mv` then it shows that your path isn't
-updated but the DVC files and/or directories are relocated to destination. So,
-using `git mv` is not an option.
+If you change your file using `mv source destination` then it shows that your
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: I still don't understand how is it relevant to DVC tbh

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 12 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: Yes, it is from an user but I think its not the latest version and I should change it.

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 14 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: I think 'relocate' means just the same as 'move'

File: static/docs/user-guide/update-DVC-pipelines.md (hunk 16 above)
Comment (Naba7) [reply static/docs/user-guide/update-DVC-pipelines.md]: I have explained it a bit more.

File: src/Documentation/sidebar.json [hunk 17]
```diff
@@ -76,7 +77,8 @@
       "autocomplete.md": "Shell Autocomplete",
       "plugins.md": "Plugins & Integrations",
       "analytics.md": "Anonymized Usage Analytics",
-      "running-dvc-on-windows.md": "Running DVC on Windows"
+      "running-dvc-on-windows.md": "Running DVC on Windows",
+      "update-DVC-pipelines.md": "Updating pipelines"
```
Comment (shcheklein) [inline src/Documentation/sidebar.json]: pipelines -> Pipelines

**Comment by Naba7** [comment]
Hi @shcheklein ! You were talking about language structure and a better layout. Any suggestions for that.

**Comment by Naba7** [comment]
ping @shcheklein

**Comment by Naba7** [comment]
ping @shcheklein

**Comment by Naba7** [comment]
Hi! @shcheklein  Can you please check this? I think its ready to be merged.

**Comment by shcheklein** [comment]
hi, @Naba7 ! sorry, have been drowned but other stuff lately. I'll try to check it asap.

**Comment by Naba7** [comment]
Sure. 😁

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 18]
```diff
@@ -0,0 +1,65 @@
+# Updating pipelines
+
+If you need to rename or relocate a DVC-file and/or dependencies and outputs
+of a specific DVC-file, then you can
+
+- use `dvc move` to edit file paths of outputs or dependencies.
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: there should be a note immediately that it's only about `dvc added` stages. They don't have dependencies. Also outputs are not always files.

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+- `dvc run` to make edits in the files and/or dependencies and outputs then
+  `dvc commit` to save changes to cache.
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: what about manually changing them?

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+
+If you want to move or rename data files and/or directories, dependencies
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: I'm not sure how is this different from the list above, tbh

File: static/docs/user-guide/update-DVC-pipelines.md [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+$ tree .
+
+.
+├── result.out
+├── stages
+│   └── test.dvc
+└── test.txt
+```
+
+If you change your file using `mv src dst` then it shows that your index isn't
```
Comment (shcheklein) [inline static/docs/user-guide/update-DVC-pipelines.md]: index? I'm not sure I understand the terminology there

**Comment by shcheklein** [comment]
Closing this as stale.

**Pull request closed without merging**